packages/adapter-mongoose/src/mongoose.ts

```diff
@@ -83,6 +83,37 @@ export const mongooseAdapter = (models: {
 				).lean();
 				return sessions.map((val) => transformSessionDoc(val));
 			},
+			getSessionAndUserBySessionId: async (sessionId: string) => {
+				if (!Session) {
+					throw new Error("Session model not defined");
+				}
+
+				const sessionUsers = await Session.aggregate([
+					{ $match: { _id: sessionId } },
+					{
+						$lookup: {
+							from: User.collection.name,
+							localField: "user_id",
+							// Relies on _id being a String, not ObjectId. 
+							// But this assumption is used elsewhere, as well
+							foreignField: "_id",
+							as: "users"
```

Comment:
I think we can omit `users: userDocs` in line 108 if we change this to `userDocs`?

Reply:
Indeed :) I've just pushed that change